sklearn/feature_selection/_correlation_threshold.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+from scipy.sparse import issparse
+from scipy.stats import spearmanr
+
+from ..base import BaseEstimator
+from .base import SelectorMixin
+from ..utils import check_array
+from ..utils.validation import check_is_fitted
+from ..utils.extmath import safe_sparse_dot
+from ..utils.sparsefuncs import min_max_axis, mean_variance_axis
+
+
+class CorrelationThreshold(BaseEstimator, SelectorMixin):
+    """Feature selector that removes features with high correlation.
+
+    This unsupervised feature selection algorithm considers feature pairs with
+    correlation higher than the `threshold` and removes the feature that has
+    the highest mean correlation with the other features. This process is
+    continued till there are no correlations higher than `threshold`.
+
+    Read more in the :ref:`User Guide <correlation_threshold>`.
+
+    Parameters
+    ----------
+    threshold : float, default=0.9
+        Features with a training-set correlation higher than this threshold
+        will be removed.
+
+    kind : {'pearson', 'spearmanr'}, default='pearson'
+        Kind or correlation to use. Dense arrays can use any kind. 'pearson' is
+        the only kind that supports sparse matrices.
+
+    Attributes
+    ----------
+    support_mask_ : bool ndarray of shape (n_features,)
+        Boolean mask for features to keep.
+
+    Examples
+    --------
+    The following example shows how one of the first highly correlated features
+    are removed, while the uncorrelated features remains:
+
+    >>> import numpy as np
+    >>> from sklearn.feature_selection import CorrelationThreshold
+    >>> X = np.array([[0.0, 1.0, 2.0], [1.1, 2.0, 3.0], [0.5, 10.1, 1.1]]).T
+    >>> selector = CorrelationThreshold(threshold=0.9)
+    >>> selector.fit_transform(X)
+    array([[ 1.1,  0.5],
+           [ 2. , 10.1],
+           [ 3. ,  1.1]])
+
+    References
+    ----------
+    Max Kuhn and Kjell Johnson, "Applied Predictive Modeling", Springer, 2013
+    (Section 3.5)
```

Comment:
you should cross link with https://scikit-learn.org/stable/modules/generated/sklearn.cluster.FeatureAgglomeration.html as it's for exact same purpose. Rather than dropping some features FeatureAgglomeration averages them.

Reply:
Done!